pkg/exec/exec.go

```diff
@@ -27,36 +28,59 @@ type OSExecutor struct {
 	DataPath string
 }
 
-const loggerKey = 1
-
 func GetLogger(ctx context.Context) log.Logger {
-	if logger, ok := ctx.Value(loggerKey).(*log.Logger); ok {
+	if logger, ok := ctx.Value(middleware.LoggerKey).(*log.Logger); ok {
 		return *logger
 	}
 
+	// Only a programming bug should cause this scenario, but the addition of
+	// this error message should at least allow us to spot a regression.
+	log.Error("Unable to retrieve logger from context, log lines will be missing")
 	return log.NewNopLogger()
 }
 
+func runCommandAndLog(logger log.Logger, message string, command *exec.Cmd) error {
```

Comment:
Would it be better to stream the output using a pipe?

Reply:
I'm not sure it's worth the extra complexity at the moment? The only real benefit of streaming that I can think of is that you'd get an indication of where in the script (e.g. `draupnir-create-new-instance`) you're at if it's frozen. But we haven't seen any issues with these scripts hanging, at least recently.
If we did stream then it'd probably have to be to a different log file, to keep it readable?

I'd propose that if we do really want this, then we tackle it in a follow-up change.

Reply:
This all makes sense- let's leave it for if we need it.